### PR TITLE
Upgrade cryptoki to 0.7.

### PR DESCRIPTION
**Pull request by partim**
This PR upgrades cryptoki – the library for PKCS11 support – to version 0.7. This requires a few minor changes related to handling of user PINs.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -27,8 +27,8 @@ basic-cookies   = { version = "^0.1", optional = true }
 bytes           = "1"
 chrono          = { version = "^0.4", features = ["serde"] }
 clap            = "^2.33"
-cryptoki        = { version = "^0.3", optional = true }
-cryptoki-sys    = { version = "=0.1.4", optional = true } # pin cryptoki-sys because of compilation issues on various systems
+cryptoki        = { version = "0.7", optional = true}
+#cryptoki-sys    = { version = "=0.1.4", optional = true } # pin cryptoki-sys because of compilation issues on various systems
```
Comment (ximon18) [inline Cargo.toml:31]: Why not delete this line rather than comment it out?

File: src/commons/crypto/signing/signers/pkcs11/signer.rs [hunk 2]
```diff
@@ -452,7 +469,7 @@ impl Pkcs11Signer {
     ) -> Result<UsableServerState, ProbeError<SignerError>> {
         fn slot_label_eq(ctx: &RwLockReadGuard<Pkcs11Context>, slot: Slot, slot_label: &str) -> bool {
             match ctx.get_token_info(slot) {
-                Ok(info) => String::from_utf8_lossy(&info.label).trim_end() == slot_label,
+                Ok(info) => info.label().trim_end() == slot_label,
```
Comment (ximon18) [inline src/commons/crypto/signing/signers/pkcs11/signer.rs:472]: It looks to me like this change potentially introduces a new panic, if the underlying token info contains a non-UTF8 compliant value. Does the `label()` call ensure that that is not possible? This change doesn't seem to be related to upgrading cryptoki or did I miss something?

**Review by ximon18 — APPROVED** [review]
LGTM, but FYI it doesn't take advantage of the changes in v0.4 of Cryptoki to resolve #901 and #902 (neither of which are critical).

File: Cargo.toml (hunk 1 above)
Comment (partim) [reply Cargo.toml:31]: That is a very good question!

**Comment by partim** [comment]
Let’s keep this PR strictly the upgrade and then do the issues in separate PRs.
